sphinx/ext/viewcode.py

```diff
@@ -50,6 +52,23 @@ def _get_full_modname(app, modname, attribute):
         return None
 
 
+def _find_module_source(app, modname):
```

Comment:
It seems this function does not uses whole of app object. How about taking `viewcode_source_dirs` as an argument instead? It makes this function much simple and testable.